### PR TITLE
DEPR: make arguments keyword only in to_dict

### PR DESCRIPTION
**Pull request by rsm-23**
- [x] xref #54229 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v2.2.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2005,25 +2009,25 @@ def to_dict(
 
         You can specify the return orientation.
 
-        >>> df.to_dict('series')
+        >>> df.to_dict(orient='series')
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py:2012]: We could also keep allowing just `orient` as positional only keyword, given how we have been using that ourselves in the docs?

Comment (rsm-23) [reply pandas/core/frame.py:2012]: @jorisvandenbossche should I remove the keyword `orient` from the examples as well?

**Comment by rsm-23** [comment]
@mroeschke should I remove the `*` and add the deprecation decorator here as well?

**Comment by mroeschke** [comment]
Yes

**Comment by rsm-23** [comment]
@jorisvandenbossche @mroeschke I am done with the changes.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rsm-23

**Pull request merged**